.github/workflows/go_tests.yaml

```diff
@@ -9,6 +9,15 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
+    services:
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ROOT_PASSWORD: githubactionpassword
+          MYSQL_DATABASE: testing_cribbage
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=5
```

Comment:
I read a lot on this in the following articles:

general service containers: https://help.github.com/en/actions/configuring-and-managing-workflows/about-service-containers
how to setup a postgres server and talk to it: https://help.github.com/en/actions/configuring-and-managing-workflows/creating-postgresql-service-containers
jobs docs: https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservices